mypy_django_plugin/main.py

```diff
@@ -306,6 +305,9 @@ def get_type_analyze_hook(self, fullname: str) -> Optional[Callable[[AnalyzeType
             return None
 
     def get_dynamic_class_hook(self, fullname: str) -> Optional[Callable[[DynamicClassDefContext], None]]:
+        if fullname == fullnames.GET_USER_MODEL_FULLNAME:
+            return partial(settings.transform_get_user_model_hook, django_context=self.django_context)
```

Comment:
Can we add branching on a helper that checks if `django.contrib.auth` is installed?

```suggestion
        if fullname == fullnames.GET_USER_MODEL_FULLNAME and self.django_context.is_contrib_auth_installed:
            return partial(settings.transform_get_user_model_hook, django_context=self.django_context)
```

That would help in the dependency hook as well

Reply:
I was intending to fix this in a separate PR. I think it's better to change the return type to `NoReturn` when auth is not configured. That's give a somewhat sensible error message, rather than `Variable "User" is not valid as a type`

Reply:
Alright, that's totally fair, we can definitely do it later on

Reply:
Note: Doing this would also assist in resolving #1380. We could include a test to resolve, looking like below, which currently crashes the plugin:

```yaml
-   case: test_xyz
    main: |
        from django.contrib.auth import get_user_model
        User = get_user_model()
    installed_apps:
        -   myapp
    files:
        -   path: myapp/__init__.py
        -   path: myapp/models.py
```